### PR TITLE
Render an empty JSON array when crud lists are empty

### PR DESCRIPTION
**Pull request by lmars**
Currently crud list endpoints return `null` rather than `[]` for empty lists, which seems wrong (I do find it surprising that `List` returns nil).

The build is currently failing with `./crud.go:77: list declared and not used`, but it is clearly "used" on line 79.

Thoughts?

### CODE REVIEW

File: crud.go [hunk 1]
```diff
@@ -72,6 +72,10 @@ func crud(resource string, example interface{}, repo Repository, r martini.Route
 			r.JSON(500, struct{}{})
 			return
 		}
+		// render an empty JSON array (rather than null) if list is nil
+		if list == nil {
+			list := make([]int, 0)
```
Comment (titanous) [inline crud.go]: This shadows list because you are using `:=`, use `=`. Also, make it a `[]struct{}`.

File: crud.go [hunk 2]
```diff
@@ -72,6 +72,10 @@ func crud(resource string, example interface{}, repo Repository, r martini.Route
 			r.JSON(500, struct{}{})
 			return
 		}
+		// render an empty JSON array (rather than null) if list is nil
+		if list == nil {
+			list = []struct{}
```
Comment (titanous) [inline crud.go]: I don't think this is syntactically valid, should be `[]struct{}{}`.

Comment (lmars) [inline crud.go]: Yeh I'm having trouble building locally so just thought I would push it up, maybe you can help with the issues:

https://gist.github.com/lmars/599be867d4188e8fdde2

I tried removing the tmpdir but it didn't help.

Comment (titanous) [inline crud.go]: Try `rm -rf /var/folders/st/x1xhp3hd22x9_lvcw9047lx00000gn/T/godep`, godep is kinda buggy.

Comment (lmars) [inline crud.go]: Yeh I tried that, no joy.

I have updated to `[]struct{}{}` but heading offline now so will look at it again tomorrow.

**Pull request merged**